### PR TITLE
Introduce `pselect`

### PR DESCRIPTION
**Pull request by giladt**
# `pselect`- dropdown

Options Data:
```
{
  value: string,
  text: string,
  innerHTML: string
}
```

Optional Parameters: 
`is-searchable` - enables search box in dropdown. Default: false
`placeholder` - placeholder text for initial state. Default: 'Please Select...'

select-dropdown

### CODE REVIEW

File: src/resources/elements/primeDesignSystem/pselect/pselect.ts [hunk 1]
```diff
@@ -0,0 +1,46 @@
+import { bindable, customElement } from "aurelia-framework";
+import SlimSelect from "slim-select";
+import "./pselect.scss";
+
+/**
+ * Usage:
+ *    <pselect type="primary">Primary</pselect>
```
Comment (dkent600) [inline src/resources/elements/primeDesignSystem/pselect/pselect.ts:7]: need to update this documentation

File: src/resources/elements/primeDesignSystem/pselect/pselect.ts [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+ *    <pselect type="tertiary" disabled>Tertiary - Disabled</pselect>
+ *    <pselect type="primary" click.delegate="message('Hi!')">Clickable</pselect>
+ *    <pselect ... full-width>Full-Width</pselect>
+*/
+
+@customElement("pselect")
+export class PButton {
+  // @bindable.booleanAttr disabled = false;
+  // @bindable.booleanAttr fullWidth = false;
+  @bindable data: Array<{value: string, text: string, innerHTML: string}>;
```
Comment (dkent600) [inline src/resources/elements/primeDesignSystem/pselect/pselect.ts:21]: Should export this as an interface like IpSelectItemConfig so consumer can get intellisense/compile errors.

File: src/resources/elements/primeDesignSystem/pselect/pselect.ts [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+  @bindable placeholder = "Please Select...";
+  @bindable isSearchable = false;
+
+  refSelectInput: HTMLSelectElement;
+  select: SlimSelect;
+  isOpen = false;
+
+  attached(): void {
+    this.select = new SlimSelect({
+      placeholder: "<span class=\"loading\"><i class=\"fas fa-circle-notch\" ></i> Loading...</span>",
```
Comment (dkent600) [inline src/resources/elements/primeDesignSystem/pselect/pselect.ts:31]: ouch more html in the typescript

File: src/resources/elements/primeDesignSystem/pselect/pselect.ts [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+      select: this.refSelectInput,
+      searchText: "DAO Name is missing.",
```
Comment (dkent600) [inline src/resources/elements/primeDesignSystem/pselect/pselect.ts:33]: This control shouldn't know anything about DAOs.  Should be agnostic.

File: src/services/DealService.ts [hunk 5]
```diff
@@ -133,11 +133,15 @@ export class DealService {
...
     const allDAOs = await(await axios.get("https://backend.deepdao.io/dashboard/ksdf3ksa-937slj3/")).data.daosSummary;
 
-    this.DAOs = allDAOs.map(dao => ({
+    this.DAOs = allDAOs.map((dao: IDaoAPIObject) => ({
       organizationId: dao.organizationId,
       daoId: dao.daoId,
       name: dao.daoName,
-      logo: `https://deepdao-uploads.s3.us-east-2.amazonaws.com/assets/dao/logo/${dao.logo}`,
+      logo: (dao.logo)
+        ? (dao.logo.includes("http"))
```
Comment (dkent600) [inline src/services/DealService.ts:141]: you mean startsWith?  case insensitive?

**Review by dkent600 — CHANGES_REQUESTED** [review]
1) should remove the yarn.lock file.  We're aligned around npm.
2) Where ever possible, would prefer if we can have less HTML in the typescript.

File: src/registry-wizard/baseStage.scss [hunk 6]
```diff
@@ -570,7 +575,8 @@
             width: -webkit-fit-content;
           }
           gap: unset;
-          input, select {
+          select,
```
[the remark this answers is not in the document]
Comment (giladt) [reply src/registry-wizard/baseStage.scss:578]: Actually, I'm not sure it's needed at all. It's part of the initial commit (means, I've copied it from launch). I won't touch it here. That can be part of the refactor of the reg wizard, IMO.

**Review by dkent600 — APPROVED** [review]

**Comment by giladt** [comment]
Moved To a new PR

**Pull request closed without merging**